src/utils/wallet.js

```diff
@@ -710,18 +721,31 @@ class Wallet {
         }
     }
 
-    async setupRecoveryMessageNewAccount(accountId, method, securityCode, fundingOptions, recoverySeedPhrase) {
+    async setupRecoveryMessageNewAccount(accountId, method, securityCode, fundingOptions, recoverySeedPhrase, recaptchaToken) {
         const { secretKey } = parseSeedPhrase(recoverySeedPhrase)
         const recoveryKeyPair = KeyPair.fromString(secretKey)
         await this.validateSecurityCode(accountId, method, securityCode);
         await this.saveAccount(accountId, recoveryKeyPair);
 
-        if (DISABLE_CREATE_ACCOUNT && !fundingOptions) {
+        // IMPLICIT ACCOUNT
+        if (DISABLE_CREATE_ACCOUNT && !fundingOptions && !recaptchaToken) {
             await store.dispatch(fundCreateAccount(accountId, recoveryKeyPair, fundingOptions, method))
             return
         }
 
-        await this.createNewAccount(accountId, fundingOptions, method, recoveryKeyPair.publicKey)
+        try {
+            // NOT IMPLICIT ACCOUNT (testnet, linkdrop, funded to delegated account via contract helper)
+            await this.createNewAccount(accountId, fundingOptions, method, recoveryKeyPair.publicKey, undefined, recaptchaToken)
+        } catch (e) {
+            if (e.code === 'NotEnoughBalance') {
```

Comment:
This is not a great kludge; feels like if we're using `dispatch()` this code belongs in an `action/xxxx` file rather than in `util`.  Will re-visit as we do wallet-SDK goodness.

Reply:
@MaximusHaximus I feel like most of stuff in `wallet` can instead be Redux actions / reducers / etc